cores/esp32/IPAddress.cpp

```diff
@@ -336,6 +210,12 @@ bool IPAddress::fromString6(const char *address) {
             colons++;
             acc = 0;
         }
+        else if (c == '%') {
```

Comment:
This looks like a simple enough change that I think it should work, but this is exactly the sort of thing it would be really nice to have unit tests for.

e.g. what happens if you try and parse "fe80::1:2:3:4%"?  Should that be an error, because the zone identifier is blank?

(A unit test could explicitly specify this by provided the expected answer).

Reply:
input fuzzing to come in the future. For now just wanted to add zone support in the already written parser (from Arduino.cc)